narwhal/primary/src/proposer.rs

```diff
@@ -315,24 +315,41 @@ impl Proposer {
     fn max_delay(&self) -> Duration {
         // If this node is going to be the leader of the next round, we set a lower max
         // timeout value to increase its chance of being included in the dag.
-        if self.committee.leader(self.round + 1).id() == self.authority_id {
+        if (self.round + 1) % 2 == 0
+            && self.leader_schedule.leader(self.round + 1).id() == self.authority_id
+        {
             self.max_header_delay / 2
         } else {
             self.max_header_delay
         }
     }
 
     fn min_delay(&self) -> Duration {
+        // TODO: consider even out the boost provided by the even/odd rounds so we avoid perpetually
+        // boosting the nodes and affect the scores.
         // If this node is going to be the leader of the next round and there are more than
         // 1 primary in the committee, we use a lower min delay value to increase the chance
-        // of committing the leader.
+        // of committing the leader. Pay attention that we use here the leader_schedule to figure out
+        // the next leader.
+        let next_round = self.round + 1;
         if self.committee.size() > 1
-            && self.committee.leader(self.round + 1).id() == self.authority_id
+            && next_round % 2 == 0
+            && self.leader_schedule.leader(next_round).id() == self.authority_id
         {
-            Duration::ZERO
-        } else {
-            self.min_header_delay
+            return Duration::ZERO;
+        }
+
+        // Give a boost on the odd rounds to a node by using the whole committee here, not just the
+        // nodes of the leader_schedule. By doing this we keep the proposal rate as high as possible
+        // which leads to higher round rate and also acting as a score booster to the less storng nodes
+        // as well.
+        if self.committee.size() > 1
+            && next_round % 2 != 0
+            && self.committee.leader(next_round).id() == self.authority_id
+        {
+            return Duration::ZERO;
         }
+        self.min_header_delay
```

Comment:
@mwtian @arun-koshy those are the changes that we'll go with to ensure that we don't change the current boosting mechanism (we'll probably change that later as follow up). I wanted to make sure that these don't go unnoticed before I merge

Reply:
IIUC there is a slight difference to the current boosting mechanism in that we are using leader schedule for the even rounds, so only certain nodes will be boosted in even rounds?

Reply:
Exactly, practically that would be the only difference here.